### DISCUSSION TITLE
Invalidation caches and max-idle

### DISCUSSION THREAD

**Discussion by vmgbrown** [infinispan/infinispan#10857] [Q&A] [answered]
The Infinispan documentation mentions that "maximum idle expiration does not work with invalidation mode":

https://infinispan.org/docs/stable/titles/configuring/configuring.html#expiration-maxidle_configuring-memory-usage

I assume this means that the `max-idle` setting has no effect on an invalidation cache? If so, I am curious to know why the default `standalone-ha.xml` configuration for WildFly (I am using v26) includes this value. Note the declaration of the "entity" cache below:

```xml
<cache-container name="hibernate" marshaller="JBOSS" modules="org.infinispan.hibernate-cache">
    <transport lock-timeout="60000"/>
    <local-cache name="local-query">
        <heap-memory size="10000"/>
        <expiration max-idle="100000"/>
    </local-cache>
    <local-cache name="pending-puts">
        <expiration max-idle="60000"/>
    </local-cache>
    <invalidation-cache name="entity">
        <heap-memory size="10000"/>
        <expiration max-idle="100000"/>
    </invalidation-cache>
    <replicated-cache name="timestamps">
        <expiration interval="0"/>
    </replicated-cache>
</cache-container>
```

If a WildFly forum would be a better place for this question, please let me know.

**Comment by wburns** [comment, marked as answer]
> I assume this means that the max-idle setting has no effect on an invalidation cache?

That is not correct, the reason it does not work is because it does not do part of what is described in the paragraph there by sending touch commands to remote nodes. So when invalidation with max idle is used each node will keep their own last access time. Thus if you have a node that doesn't access the entry it will send an invalidation message removing it from all nodes, causing a refetch of the data.

Note, however that Wildfly clustering implements their own max idle on top of Infinispan. You will want to ask them what semantics it has.

**Reply by vmgbrown** [reply]
Thanks for the information - that makes sense. You may want to elaborate on that in the documentation. As written, it implies that `max-idle` is not supported by invalidation caches.
